crates/hir-expand/src/db.rs

```diff
@@ -338,13 +339,26 @@ pub(crate) fn parse_with_map(
         }
     }
 }
+/// This is just to ensure the types of smart_macro_arg and macro_arg are the same
+type MacroArgResult = (Arc<tt::Subtree>, SyntaxFixupUndoInfo, Span);
+/// Imagine the word smart in quotes.
+///
+/// This resolves the [MacroCallId] to check if it is a derive macro if so get the [macro_arg] for the derive.
+/// Other wise return the [macro_arg] for the macro_call_id.
+///
+/// This is not connected to the database so it does not cached the result. However, the inner [macro_arg] query is
+fn macro_arg_considering_derives(db: &dyn ExpandDatabase, id: MacroCallId) -> MacroArgResult {
+    let macro_call_kind = db.lookup_intern_macro_call(id).kind;
+    // FIXME: We called lookup_intern_macro_call twice.
```

Comment:
Just add the `MacroCallKind` as an additional param here, a bit ugly but that's not really too problematic in comparison to the extra lookup